crates/yulgen/src/runtime/functions/contracts.rs

```diff
@@ -59,12 +51,12 @@ pub fn calls(contract: Contract) -> Vec<yul::Statement> {
                     }
                 }
             } else {
-                let decoding_operation = abi_operations::decode(
-                    vec![function.return_type],
+                let decoding_operation = abi_operations::decode_data(
+                    &[function.return_type],
+                    identifier_expression! { outstart },
                     identifier_expression! { outstart },
```

Comment:
this should be `outstart + outsize`. it's not an issue rn since the `data_end` parameter is ignored, so Im just going to fix it in #440